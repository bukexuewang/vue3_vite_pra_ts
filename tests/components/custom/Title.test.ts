import { mount } from '@vue/test-utils';
import TitleVue from '@/components/custom/Title.vue';

describe('测试vue组件', () => {
  it('展示title组件，小属性的情况', async () => {
    const wrapper = mount(TitleVue, {
      props: {
        title: 'Hello world',
        type: 'small',
      },
    });

    // Assert the rendered text of the component
    // expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('p').text()).toBe('0');
    await wrapper.find('button').trigger('click');
    expect(wrapper.find('p').text()).toBe('1');
  });

  // it('展示title组件，大属性的情况', () => {
  //   const wrapper = mount(TitleVue, {
  //     props: {
  //       title: 'Hello world',
  //       type: 'large',
  //     },
  //   });

  //   // Assert the rendered text of the component
  //   expect(wrapper.html()).toMatchSnapshot();
  // });
});
