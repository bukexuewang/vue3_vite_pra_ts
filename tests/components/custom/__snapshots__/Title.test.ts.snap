// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group displays message 1`] = `"<div class=\\"title\\" style=\\"font-size: 0.5em; color: green;\\">Hello world</div>"`;

exports[`测试vue组件 展示title组件，大属性的情况 1`] = `"<div class=\\"title\\" style=\\"font-size: 2em; color: red;\\">Hello world</div>"`;

exports[`测试vue组件 展示title组件，小属性的情况 1`] = `"<div class=\\"title\\" style=\\"font-size: 0.5em; color: green;\\">Hello world</div>"`;
